system-tests/lib/cre/features/ocr/ocr.go

```diff
@@ -0,0 +1,528 @@
+package ocr
+
+import (
+	"context"
+	"fmt"
+	"sort"
+
+	"github.com/Masterminds/semver/v3"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/pkg/errors"
+	"github.com/rs/zerolog"
+	"golang.org/x/sync/errgroup"
+
+	chainselectors "github.com/smartcontractkit/chain-selectors"
+
+	capabilitiespb "github.com/smartcontractkit/chainlink-common/pkg/capabilities/pb"
+	kcr "github.com/smartcontractkit/chainlink-evm/gethwrappers/keystone/generated/capabilities_registry_1_1_0"
+	"github.com/smartcontractkit/chainlink-protos/job-distributor/v1/job"
+	"github.com/smartcontractkit/chainlink-testing-framework/framework/clclient"
+
+	"github.com/smartcontractkit/chainlink-deployments-framework/datastore"
+	cldf "github.com/smartcontractkit/chainlink-deployments-framework/deployment"
+	"github.com/smartcontractkit/chainlink-deployments-framework/offchain"
+	"github.com/smartcontractkit/chainlink-deployments-framework/operations"
+	"github.com/smartcontractkit/chainlink/deployment/cre/ocr3"
+	keystone_changeset "github.com/smartcontractkit/chainlink/deployment/keystone/changeset"
+	ks_contracts_op "github.com/smartcontractkit/chainlink/deployment/keystone/changeset/operations/contracts"
+
+	libc "github.com/smartcontractkit/chainlink/system-tests/lib/conversions"
+	"github.com/smartcontractkit/chainlink/system-tests/lib/cre"
+	"github.com/smartcontractkit/chainlink/system-tests/lib/cre/contracts"
+	"github.com/smartcontractkit/chainlink/system-tests/lib/cre/don/jobs"
+	"github.com/smartcontractkit/chainlink/system-tests/lib/cre/flags"
+	"github.com/smartcontractkit/chainlink/system-tests/lib/infra"
+)
+
+const flag = cre.ConsensusCapability
+
+type OCR struct{}
+
+func (o *OCR) Flag() cre.CapabilityFlag {
+	return flag
+}
+
+func (o *OCR) PreDONStartup(
+	testLogger zerolog.Logger,
+	registryChainSelector uint64,
+	cldfEnv *cldf.Environment,
+	provider infra.Provider,
+	topology *cre.Topology,
+	blockchainOutputs []*cre.WrappedBlockchainOutput,
+	capabilityConfigs cre.CapabilityConfigs,
+	contractVersions map[string]string,
+) error {
+	// nothing to do
+	return nil
+}
+
+const (
+	OCR3ContractQualifier    = "capability_ocr3"
+	DONTimeContractQualifier = "capability_dontime"
+)
+
+func (o *OCR) PostDONStartup(
```

Comment:
@Tofel We're tying stuff together here -- The OCR3 capability  and DonTime are separate "features"; they should have separate set ups

Reply:
Will untangle, this was copy pasta